api/src/Shipment/ShippingService.php

```diff
@@ -9,18 +9,43 @@
 class ShippingService
 {
     private $shipping_api_url;
+    private $shipping_app_url;
     private $headers;
     public const JOURNEY_TO_FACILITY = "TO_FACILITY";
     public const JOURNEY_FROM_FACILITY = "FROM_FACILITY";
 
-    function __construct()
+    function _build_headers()
     {
-        global $shipping_service_url;
-        $this->shipping_api_url = $shipping_service_url;
-        $this->headers = array(
+        global $cookie_key;
+        global $shipping_service_api_user;
+        global $shipping_service_api_password;
+
+        $headers = array(
             'Accept: application/json',
             'Content-Type: application/json',
         );
+
+        if (isset($_COOKIE[$cookie_key])) {
+            array_push($headers, "Authorization: Bearer {$_COOKIE[$cookie_key]}");
+            return $headers;
+        }
+
+        if (isset($shipping_service_api_user) && isset($shipping_service_api_password)) {
+            $basic_auth = base64_encode($shipping_service_api_user . ":" . $shipping_service_api_password);
+            array_push($headers, "Authorization: Basic {$basic_auth}");
+            return $headers;
+        }
+
+        throw new \Exception("Shipping service auth error: no cookie found and basic auth credentials unset");
+    }
+
+    function __construct()
```

Comment:
Constructing this object will fail if the setting are not set and might fail if they are null. Either:
- make sure this is not constructed if the shipping service is not being used
- make sure the constructor doesn't do anything if the shipping service is not used